src/connector/src/sink/postgres.rs

```diff
@@ -506,7 +510,13 @@ impl PostgresSinkWriter {
                 schema.fields(),
             );
             let statement = match op {
-                Op::Insert => create_insert_sql(schema, table_name, rows_length),
+                Op::Insert => {
+                    if append_only {
+                        create_insert_sql(schema, table_name, rows_length)
+                    } else {
+                        create_upsert_sql(schema, table_name, pk_indices)
+                    }
```

Comment:
I am not familiar with how it works previously, but these 2 branches look imbalanced:
* `create_insert_sql` generates `format!("${}", i * number_of_columns + j + 1)`
* `create_upsert_sql` generates `format!("${}", i + 1)`

Reply:
And FYI PostgreSQL supports a special table name `excluded`:
```
ON CONFLICT (did) DO UPDATE SET dname = EXCLUDED.dname;
```
https://www.postgresql.org/docs/current/sql-insert.html
This can be helpful when handling multi-row insertion.

Another issue not limited to this PR: **interpolating SQL string is error-prone**. For example, the table and column names shall be quoted (`"foo"`) or uppsercase letters are folded to lowercase by PostgreSQL - more tricky is that quotes shall be doubled (e.g. when `field.name` is `abC"Def` the correct quoting is `"abC""Def"`)

Reply:
> And FYI PostgreSQL supports a special table name excluded:
ON CONFLICT (did) DO UPDATE SET dname = EXCLUDED.dname;
https://www.postgresql.org/docs/current/sql-insert.html
This can be helpful when handling multi-row insertion.

Thanks, this is helpful.

> Another issue not limited to this PR: interpolating SQL string is error-prone. For example, the table and column names shall be quoted ("foo") or uppsercase letters are folded to lowercase by PostgreSQL - more tricky is that quotes shall be doubled (e.g. when field.name is abC"Def the correct quoting is "abC""Def")

Yeah I guess we need to audit PG CDC as well for this. For PG Native Sink, I don't think we have that quoting in place.

Reply:
Fixed